skimage/viewer/__init__.py

```diff
@@ -1,6 +1,14 @@
 from .._shared.utils import warn
-from .viewers import ImageViewer, CollectionViewer
+from .._shared import has_mpl
 from .qt import has_qt
 
 if not has_qt:
     warn('Viewer requires Qt')
+
+if has_mpl:
+    from .viewers import ImageViewer, CollectionViewer
+else:
+    ImageViewer, CollectionViewer = None, None
+    warn('Viewer requires matplotlib.', stacklevel=2)
```

Comment:
```suggestion
    warn('Viewer requires matplotlib', stacklevel=2)
```
(just because messages end with no period in the rest of file and in the rest of the PR)